### DISCUSSION TITLE
class_loss_weights for Object Detection

### DISCUSSION THREAD

**Discussion by AashishKumbashi** [azavea/raster-vision#2022] [Q&A]
Since class_loss_weights is not supported for object detection tasks by rastervision, is there any other way in which I can handle class imbalance problem for object detection?

**Comment by AdeelH** [comment]
If you have just one target class, you can use the [`neg_ratio` argument](https://docs.rastervision.io/en/0.21/api_reference/_generated/rastervision.pytorch_learner.dataset.object_detection_dataset.ObjectDetectionRandomWindowGeoDataset.html#id0) in `ObjectDetectionRandomWindowGeoDataset`. 

There isn't a built-in mechanism for dealing with multiple classes. You might consider subclassing and modifying `ObjectDetectionRandomWindowGeoDataset` to accept a list of per-class sampling probabilities instead of `neg_ratio` and sample according to those.
